servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java

```diff
@@ -124,9 +124,12 @@ public Reference createReference(@Nullable String sourceRefName, Reference refer
   @Override
   protected void assignReference(NamedRef ref, String oldHash, Reference assignTo)
       throws NessieNotFoundException, NessieConflictException {
-    namedRefWithHashOrThrow(assignTo.getName(), assignTo.getHash());
-
-    getAccessChecker().canAssignRefToHash(createAccessContext(), ref);
+    ServerAccessContext accessContext = createAccessContext();
+    getAccessChecker()
+        .canViewReference(
+            accessContext,
+            namedRefWithHashOrThrow(assignTo.getName(), assignTo.getHash()).getValue());
+    getAccessChecker().canAssignRefToHash(accessContext, ref);
```

Comment:
I think we need to handle similarly for `transplant` also. Better to check all the API once and handle it as mentioned in #3058

Reply:
indeed, we need to check merge+transplant

Reply:
yes, please

Reply:
fixed + added tests